test/infamy/transport.py

```diff
@@ -54,6 +54,23 @@ def get_mgmt_iface(self):
         """Return managment interface used for RESTCONF/NETCONF"""
         return self.location.interface
 
+    def has_model(self, model_name, xpath=None):
+        """
+        Check if the device has the given YANG model loaded.
+
+        :param model_name: Name of the YANG model to check.
+        :param xpath: Optional XPath to refine search to a specific property.
+        :return: True if the model is found, False otherwise.
+        """
+        # If xpath is provided, extract the module name from it
+        if xpath:
+            mod = [seg.split(":")[0] for seg in xpath.split("/") if ":" in seg]
+            if model_name not in mod:
+                return False
```

Comment:
I can't wrap my head around how this is supposed to be used - and it seems there are no users in this PR at least. Do we need it?

Reply:
It was for checking for features in the model, not used atm., but since we don't look for revision I though this could come in handy.

Reply:
Removed in my local queue.